include/seqan3/io/alignment_file/input.hpp

```diff
@@ -0,0 +1,1323 @@
+     * \param[in] format_tag The file format tag.
+     * \param[in] fields_tag A seqan3::fields tag. [optional]
+     *
+     * \details
+     *
+     * ### Decompression
+     *
+     * This constructor transparently applies a decompression stream on top of the stream in case
+     * it is detected as being compressed.
+     * See the section on \link io_compression compression and decompression \endlink for more information.
+     */
+    template <IStream2 stream_t, AlignmentFileInputFormat file_format>
+    alignment_file_input(stream_t                 & stream,
+                         file_format        const & SEQAN3_DOXYGEN_ONLY(format_tag),
+                         selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{&stream, stream_deleter_noop}, format{file_format{}}
+    {
+        static_assert(meta::in<valid_formats, file_format>::value,
+                      "You selected a format that is not in the valid_formats of this file.");
+
+        // possibly add intermediate compression stream
+        secondary_stream = detail::make_secondary_istream(*primary_stream);
+
+        // buffer first record
+        read_next_record();
+    }
+
+    //!\overload
+    template <IStream2 stream_t, AlignmentFileInputFormat file_format>
+    alignment_file_input(stream_t                && stream,
+                         file_format        const & SEQAN3_DOXYGEN_ONLY(format_tag),
+                         selected_field_ids const & SEQAN3_DOXYGEN_ONLY(fields_tag) = selected_field_ids{}) :
+        primary_stream{new stream_t{std::move(stream)}, stream_deleter_default},
+        format{file_format{}}
+    {
+        static_assert(meta::in<valid_formats, file_format>::value,
+                      "You selected a format that is not in the valid_formats of this file.");
+
+        // possibly add intermediate compression stream
+        secondary_stream = detail::make_secondary_istream(*primary_stream);
+
+        // buffer first record
+        read_next_record();
+    }
+
+    /*!\brief Construct from filename and give additional reference information.
+     * \tparam    ref_ids_t       The range type that stores the reference ids.
+     * \tparam    ref_sequences_t The range type that stores the reference information.
+     * \param[in] filename        Path to the file you wish to open.
+     * \param[in] ref_ids         A range containing the reference ids that correspond to the SAM/BAM file.
+     * \param[in] ref_sequences   A range containing the reference sequences that correspond to the SAM/BAM file.
+     * \param[in] fields_tag      A seqan3::fields tag. [optional]
+     * \throws seqan3::file_open_error If the file could not be opened, e.g. non-existant, non-readable, unknown format.
+     *
+     * \details
+     *
+     * In addition to the file name and the reference information, you may specify a custom seqan3::fields type which
```

Comment:
Same here, you give a value of the type of a  custom seqan3::fields  and not the type itself